pkg/resources/services/service.go

```diff
@@ -151,7 +151,11 @@ func ServiceForBucket(t *miniov2.Tenant, bucket string) *corev1.Service {
 
 // NewHeadlessForMinIO will return a new headless Kubernetes service for a Tenant
 func NewHeadlessForMinIO(t *miniov2.Tenant) *corev1.Service {
-	minioPort := corev1.ServicePort{Port: miniov2.MinIOPort, Name: miniov2.MinIOServiceHTTPPortName}
+	name := miniov2.MinIOServiceHTTPPortName
+	if t.TLS() {
+		name = miniov2.MinIOServiceHTTPSPortName
```

Comment:
The code looks fine and certainly is an improvement for new deployments, but what happens with upgrades? I see this method is invoked from `controller.checkMinIOHLSvc(...)` and the result is compared to the actual service. If it doesn't match, then the existing service is updated. I guess this means that for existing services, the port is renamed from `http-minio` to `https-minio`. This may impact installations and should be mentioned in upgrade documentation.

Reply:
Good point. Thanks @ramondeklein

Reply:
Yes. According to this
```
	// The name of this port within the service. This must be a DNS_LABEL.
	// All ports within a ServiceSpec must have unique names. When considering
	// the endpoints for a Service, this must match the 'name' field in the
	// EndpointPort.
	// Optional if only one ServicePort is defined on this service.
	// +optional
	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
```
That will change the `EndpointPort`.
I think that's ok for no one will use that port. @ramondeklein

Reply:
For another customer that I worked for, I introduced an `UPGRADE.md` file in all (public) repos. Whenever you make a change that may have impact during upgrades, then it was mandatory to describe the change (including a link to the PR) in that file.

Right before a new version was released, the version tag was added to the file. When there were (possible) breaking changes then we updated the minor number. If there were no breaking changes (that we knew of), then we updated the patch number. It might be a good idea to do that for MinIO too.

It's very frustrating for users if deployments break after upgrade. Not everyone will quickly find out that the name of a port has been changed.

Reply:
@jiuker The name doesn't change for the internal DNS, because headless services will be named `<service>.<namespace>.svc.<cluster-domain>` and because the name of the headless service doesn't change, this won't impact internal DNS resolving. I also checked the `Ingress` resource and it also refers to the port number and not the service's port name. So it probably won't break a lot of installations.

Reply:
Yes. Good idea. @ramondeklein 
Have a plan for this? @pjuarezd

Reply:
> @jiuker The name doesn't change for the internal DNS, because headless services will be named `<service>.<namespace>.svc.<cluster-domain>` and because the name of the headless service doesn't change, this won't impact internal DNS resolving. I also checked the `Ingress` resource and it also refers to the port number and not the service's port name. So it probably won't break a lot of installations.

Yeah. That's why I don't think it's a really broken change here. @ramondeklein

Reply:
It's ok, @ramondeklein is right, this is a potential breaking change worth to documment and advise, here will take track of this documentation needs https://github.com/minio/operator/issues/2136,  but I really think this PR is a good thing to include, makes services port names consistent.